LIPS/lip-3.md

```diff
@@ -0,0 +1,152 @@
+
+![Easy Track system schema](./assets/lip-3/easytrackschema.jpg)
+___
+
+## Core smart contracts
+
+The list of core Easy Track contracts includes:
+- `EasyTrack` - contains the basic motion logic and data related to all the active Easy Track motions;
+- `MotionSettings` - contains logic to set objections threshold, motion duration and maximum active motions count;
+- `EVMScriptFactoriesRegistry` - contains logic to add/remove `EVMScriptFactory` contracts and also an internal method to create EVM scripts;
+- `EasyTrackStorage` - inherited by `MotionSettings` and `EVMScriptFactoriesRegistry` to simplify Easy Track upgradability by storing all the variables in a single contract.
+
+### `EasyTrack` contract
+
+`EasyTrack` smart contract is the main contract containing the basic motion logic and data related to all the ongoing (active) Easy Track motions, including the enactment EVM script (hashed), the address that has started the motion, and when it has been started. 
+We propose using byte data type for calldata due to various sets of parameters required for different motion types.
+`EasyTrack` stores all the active motions (the ones created but neither enacted nor canceled yet). Once enacted, rejected, or canceled, the motion gets permanently deleted from the smart contract storage, because it is not possible to interact with past motions in any way. All the interactions with active motions also happen as transactions to EasyTrack smart contract, i.e.:
+- create motion;
+- cancel motion;
+- submit objection;
+- enact motion.
+
+### `MotionSettings` contract
+
+`MotionSettings` contract provides methods to control easy track settings, including motion duration, objections threshold and maximum active motions count. Inherits from `EasyTrackStorage`.
+
+### `EVMScriptFactoriesRegistry` contract
+
+`EVMScriptFactoriesRegistry` contract provides methods to add/remove `EVMScriptFactory` into the list of allowed `EVMScriptFactory` contracts and also an internal method to create EVM scripts. `EVMScriptFactoriesRegistry` inherits from `EasyTrackStorage`.
+The internal method `_createEVMScript()` creates the enactment EVM script by calling the corresponding `EVMScriptFactory` contract and also checks if the resulting script meets all the required script factory permissions.
+
+### `EasyTrackStorage` contract
+
+`EasyTrackStorage` contract stores all the variables of Easy Track system. It inherits from battle-tested `Initializable`, `PausableUpgradeable` and `AccessControlUpgradeable` OpenZeppelin contracts.
+On the other hand, `MotionSettings` and `EVMScriptFactories` inherit from `EasyTrackStorage` to make variables stored in the latter accessible to their methods.
+
+______
+## `EVMScriptFactory` smart contracts
+
+Initially, five `EVMScriptFactory` contracts will represent five Easy Track motion types.
```

Comment:
We've mentioned three types on the start; I know that we'll have all five, but here it's worth explaining the difference, imo

Reply:
I don't see a contradiction here, as we mentioned three types of votings we would like to wrap into motions. Wrapping resulted in five motions, which is pretty obvious from the list right below this line.